gcp/api/server.py

```diff
@@ -222,14 +219,16 @@ def _is_version_affected(affected_packages,
                          purl,
                          version,
                          normalize=False):
-  """Returns whether or not the given version is within an affected ECOSYSTEM
-  range."""
+  """Returns whether or not the given version is within an affected ECOSYSTEM range."""
   for affected_package in affected_packages:
     if package_name and package_name != affected_package.package.name:
       continue
 
-    if ecosystem and ecosystem != affected_package.package.ecosystem:
-      continue
+    if ecosystem:
+      # If package ecosystem has a :, also try ignoring parts after it.
+      if affected_package.package.ecosystem != ecosystem and affected_package.package.ecosystem.split(
```

Comment:
nit: I think this line is long 

side note: I think our linting is broken and not picking this up properly.

Reply:
we can simplify this a bit:

```python
normalized_ecosystem = affected_package.package.ecosystem.split(':')[0]
if normalized_ecosystem != ecosystem:
  continue
```

And actually, since we have a number of places where we are doing this ':' splitting (e.g. in `models.py`), let's just add a helper function to `ecosystems.py` called `normalize` which doese this splitting.

Reply:
Wouldn't this fail if someone searches for `Debian:11`, instead of just `Debian`?

Reply:
Good point! 

We should be doing this then:

```python
if ecosystems.normalize(affected_package.package.ecosystem) != ecosystems.normalize(ecosystem):
  continue
```